cluster/ubuntu/util.sh

```diff
@@ -770,6 +769,33 @@ function kube-down() {
   done
 }
 
+# Clean up the k8s-resources before kube-down
+#     1) If master need down, cleaned up all resources
+#     2) If only nodes need down, tansfer pods and cleaned up node-resources
+function prepare-kube-down() {
+  local delete_master=false
+  for role in ${roles_array}; do
+    if [[ "${role}" == "ai" || "${role}" == "a" ]]; then
+      delete_master=true
+    fi
+  done
+  
+  if ${delete_master}; then
+    tear_down_alive_resources
+    check-pods-torn-down
+  else
+    # Just delete node-resources
+    local kubectl="${KUBE_ROOT}/cluster/kubectl.sh"
+    for i in ${nodes}; do
+      to_be_deleted_node=${i#*@}
+      echo "Down node ${to_be_deleted_node} from cluster"
+	  "${kubectl}" drain ${to_be_deleted_node}
+      "${kubectl}" delete node ${to_be_deleted_node}
+    done
+    echo "Nodes release resources, please wait 15 seconds ..."
+    sleep 15
+  fi
+}
```

Comment:
As I have reply, my codes will be work when "master" not defined in $roles, so process will be able to perform normally even if "master" not defined. 
For example:
I have installed cluster enviroment include master(root@10.0.0.1) and node(root@10.0.0.2):
	a) The follwing procedures can help us add the other node(root@10.0.0.3) into this cluster:
		1) config nodes info(file: "config-default.sh")
			nodes="root@10.0.0.3"
			roles="i"
			NUM_NODES=1
			MASTER="root@10.0.0.1"
		2) execute kube-up.sh
	b) The follwing procedures can help us remove node(root@10.0.0.2) from this cluster:
		1) config nodes info(file: "config-default.sh")
			nodes="root@10.0.0.2"
			roles="i"
			NUM_NODES=1
			MASTER="root@10.0.0.1"
		2) execute kube-down.sh
According to the above process, nodes can be adjusted dynamically without affecting the whole exist cluster.